stories/features/shields/components/blockedReources/dynamicList.tsx

```diff
@@ -0,0 +1,182 @@
+      if (item.blocked === false) {
+        return null
+      }
+      return (
+        <ResourcesListGrid hightlighted={item.hasUserInput} key={index}>
+          {
+            item.hasUserInput
+            ? <ResourcesListBlockedLink>{locale.blocked}</ResourcesListBlockedLink>
+            : <Link data-item={item.index} onClick={this.onClickAllowItem}>{locale.allow}</Link>
+          }
+          <ResourcesListItem>{item.name}</ResourcesListItem>
+          {
+            item.hasUserInput
+            ? <Link onClick={this.onClickUndoAction}>{locale.undo}</Link>
+            : null
+          }
+        </ResourcesListGrid>
+      )
+    })
+  }
+
+  disabledList = (list: any[]) => {
+    return list.map((item, index) => {
+      if (item.blocked === true) {
+        return null
+      }
+      return (
+        <ResourcesListGrid hightlighted={item.hasUserInput} key={index} >
+          {
+            item.hasUserInput
+            ? <ResourcesListAllowedLink>{locale.allowed}</ResourcesListAllowedLink>
+            : <Link onClick={this.onClickBlockItem}>{locale.block}</Link>
+          }
+          <ResourcesListItem>{item.name}</ResourcesListItem>
+          {
+            item.hasUserInput
+            ? <Link onClick={this.onClickUndoAction}>{locale.undo}</Link>
+            : null
+          }
+        </ResourcesListGrid>
+      )
+    })
+  }
+
+  render () {
+    const { showApplyOptions } = this.state
+    const { list, onClickDismiss } = this.props
+    return (
+      <>
+        <ResourcesListScroll>
+          {showApplyOptions ? <DismissOverlay onClick={this.onClickOutsideBounds} /> : null}
+          {/* blocked scripts */}
+          <ResourcesSubTitleGrid>
+            <BlockedScriptsIcon />
+            <ResourcesSubTitle accent='blocked'>
+              {locale.blockedScripts} ({this.blockedListSize})
+            </ResourcesSubTitle>
+            <Link>{locale.allowAll}</Link>
+          </ResourcesSubTitleGrid>
+          {this.enabledList(list)}
+
+          {/* allowed scripts */}
+          <ResourcesSubTitleGrid>
+            <AllowedScriptsIcon />
+            <ResourcesSubTitle accent='allowed'>
+              {locale.allowedScripts} ({this.allowedListSize})
+            </ResourcesSubTitle>
+            <Link>{locale.blockAll}</Link>
+          </ResourcesSubTitleGrid>
+          {this.disabledList(list)}
+
+        </ResourcesListScroll>
+        <ResourcesFooterGrid>
+          <ResourcesFooterGridColumnLeft>
+            <Link onClick={onClickDismiss}>{locale.cancel}</Link>
+          </ResourcesFooterGridColumnLeft>
+          <ResourcesFooterGridColumnRight>
+            <Button
+              level='primary'
+              type='accent'
+              text={locale.apply}
+              onClick={this.onClickApplyOptions}
+              icon={{ position: 'after', image: <CaratDownIcon /> }}
+            />
+            <Options visible={showApplyOptions} position={{ bottom: '58px' }}>
```

Comment:
This is the only place in this PR where we're specifying the style of an item outside of the `style.ts` file. Is there a semantic property we could use so that this `bottom: 58px` is defined alongside this component's definition, keeping all the style for it in one place? I can't seem to see where in the storybook this item is, so I can't provide a suggestion myself 🤷‍♂️

Reply:
plan was to make it a shared component in the future but let's do one battle at a time. changed to the hard-coded version

Reply:
Sure a prop is ok, I was wondering if the exact bottom: 58px can be controlled inside the style rather than the render